web/nginx/nginx.conf

```diff
@@ -0,0 +1,85 @@
+# https://nginxconfig.io
+# https://www.nginx.com/resources/wiki/start/topics/examples/full/
+
+user nginx;
+pid /run/nginx.pid;
+#worker_processes  1;
+worker_processes auto;
+worker_rlimit_nofile 1024;
+
+# Keeps the logs free of messages about not being able to bind().
+# #daemon     off;
+
+#include /etc/nginx/modules-enabled/*.conf;
+#load_module "modules/ngx_pagespeed.so";
+
+events {
+  multi_accept on;
+  #worker_connections 65535;
+  worker_connections 1024;
+}
+
+
+http {
+  charset utf-8;
+  sendfile on;
+  tcp_nopush on;
+  tcp_nodelay on;
+  server_tokens off;
+  log_not_found off;
+  types_hash_max_size 2048;
+  client_max_body_size 16M;
+
+  keepalive_timeout  65;
+
+  # Log
+  log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                    '$status $body_bytes_sent "$http_referer" '
+                    '"$http_user_agent" "$http_x_forwarded_for"';
+  access_log  /var/log/nginx/nginx-access.log  main;
+  error_log  /var/log/nginx/nginx-error.log warn;
+
+  # MIME
+  include mime.types;
+  default_type application/octet-stream;
+
+  # SSL
+  ssl_session_timeout 1d;
+  ssl_session_cache shared:SSL:50m;
+  ssl_session_tickets off;
+
+  # modern configuration
+  ssl_protocols TLSv1.2;
+  # ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+  ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256;
+  # ssl_ciphers "ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS";
+  ssl_prefer_server_ciphers on;
+
+  # OCSP Stapling
+  ssl_stapling on;
+  ssl_stapling_verify on;
+  resolver 1.1.1.1 1.0.0.1 8.8.8.8 8.8.4.4 208.67.222.222 208.67.220.220 valid=60s;
+  resolver_timeout 2s;
+
+  # https://developer.mozilla.org/en-US/docs/HTTP/X-Frame-Options
+  add_header X-Frame-Options SAMEORIGIN;
+  # http://msdn.microsoft.com/en-us/library/ie/gg622941(v=vs.85).aspx
+  add_header X-Content-Type-Options nosniff;
+  # https://www.owasp.org/index.php/List_of_useful_HTTP_headers
+  add_header X-XSS-Protection "1; mode=block";
```

Comment:
I like this, are there any more of these

Reply:
Nice catch. They are actually duplicated in general.conf. Fixed

`general.conf` is the conf snippet which should be generally included in all the individual site conflagrations.

I will try to search nginx.conf in GitHub. But as you know, in this tutorial-driven-development world, it's hard. I have included what I can find. Maybe add new ones in the future.